src/main/java/uk/gov/hmcts/reform/sscs/config/CitizenCcdService.java

```diff
@@ -78,6 +81,29 @@ public SaveCaseResult saveCase(SscsCaseData caseData, IdamTokens idamTokens) {
         }
     }
 
+    public SaveCaseResult saveCaseV2(IdamTokens idamTokens, Consumer<SscsCaseData> mutator) {
+
+        List<CaseDetails> caseDetailsList = citizenCcdClient.searchForCitizen(idamTokens);
+
+        CaseDetails caseDetails;
+
+        if (CollectionUtils.isNotEmpty(caseDetailsList)) {
+
+            String caseId = caseDetailsList.get(0).getId().toString();
```

Comment:
you can use findFirst of stream api

Reply:
Hi @nitinprabhuhmcts this section of code is lifted from the existing save case function so I didn't want to adapt any logic